codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -0,0 +1,403 @@
+                writer.write("entryDecoder := smithydecoding.NewXMLNodeDecoder(decoder.Decoder, t)");
+                // delegate to unwrapped map deserializer function
+                writer.openBlock("if err := $L(&sv, entryDecoder); err != nil {", "}",
+                        getUnwrappedMapDelegateFunctionName(context, shape), () -> {
+                    writer.write("return err");
+                    });
+
+                writer.openBlock("default:", "", () -> {
+                    writer.write("// Do nothing and ignore");
+                });
+            });
+        });
+        writer.write("*v = sv");
+        writer.write("return nil");
+    }
+
+    // Generates deserializer function for flattened maps.
+    protected void generateFlattenedMapDeserializer(GenerationContext context, MapShape shape) {
+        GoWriter writer = context.getWriter();
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        Symbol symbol = symbolProvider.toSymbol(shape);
+
+        writer.openBlock("func $L(v *$P, decoder *smithydecoding.XMLNodeDecoder) error {", "}",
+                getUnwrappedMapDelegateFunctionName(context, shape), symbol, () -> {
+                    // Initialize the value now that the start stub has verified that there's something there.
+                    writer.write("var sv $P", symbol);
+                    writer.openBlock("if *v == nil {", "", () -> {
+                        writer.write("sv = make($P, 0)", symbol);
+                        writer.openBlock("} else {", "}", () -> {
+                            writer.write("sv = *v");
+                        });
+                    });
+
+                    writer.write("var ek $P", symbolProvider.toSymbol(shape.getKey()));
+                    writer.write("var ev $P", symbolProvider.toSymbol(shape.getValue()));
+                    writer.insertTrailingNewline();
+
+                    // Iterate through the decoder. The member visitor will handle popping tokens.
+                    writer.openBlock("for {", "}", () -> {
+                        writer.write("t, done, err := decoder.Token()");
+                        writer.write("if err != nil { return err }");
+                        writer.openBlock("if done {", "}", () -> {
+                            // set the key value pair in map
+                            writer.write("sv[*ek] = ev");
+                            writer.write("break");
+                        });
+
+                        writer.openBlock("switch t.Name.Local {", "}", () -> {
+                            MemberShape keyShape = shape.getKey();
+                            writer.openBlock("case $S:", "", getSerializedMemberName(keyShape), () -> {
+                                String dest = "ek";
+                                context.getModel().expectShape(keyShape.getTarget()).accept(getMemberDeserVisitor(keyShape, dest,((shape1)-> !shape1.hasTrait(XmlAttributeTrait.ID))));
```

Comment:
This looks like more than 120 characters. Also lambdas with one argument can be be set up without parens, e.g. `s -> s.foo` vs `(s) -> s.foo`. Also, only structure members can have the `xmlAttribute` trait.